openupgrade_scripts/scripts/account/15.0.1.2/pre-migration.py

```diff
@@ -0,0 +1,409 @@
+from openupgradelib import openupgrade
+
+from odoo.tools import sql
+
+_renamed_fields = [
+    (
+        "account.move",
+        "account_move",
+        "tax_cash_basis_move_id",
+        "tax_cash_basis_origin_move_id",
+    ),
+    (
+        "account.reconcile.model",
+        "account_reconcile_model",
+        "match_total_amount_param",
+        "payment_tolerance_param",
+    ),
+    (
+        "account.reconcile.model.template",
+        "account_reconcile_model_template",
+        "match_total_amount_param",
+        "payment_tolerance_param",
+    ),
+    (
+        "account.reconcile.model",
+        "account_reconcile_model",
+        "match_total_amount",
+        "allow_payment_tolerance",
+    ),
+    (
+        "account.reconcile.model.template",
+        "account_reconcile_model_template",
+        "match_total_amount",
+        "allow_payment_tolerance",
+    ),
+]
+
+
+def switch_allow_payment_tolerance_value(env):
+    for table in ["account_reconcile_model", "account_reconcile_model_template"]:
+        openupgrade.logged_query(
+            env.cr,
+            f"""
+            UPDATE {table}
+            SET allow_payment_tolerance = NOT allow_payment_tolerance
```

Comment:
If you check the diff between Odoo 14 and Odoo 15, it looks like the values of `allow_payment_tolerance` remain unchanged compared to the old `match_total_amount`, and it's the `payment_tolerance_param` that needs to be inversed (as `100 - match_total_amount_param`). This is most obvious in the view diff:

![image](https://user-images.githubusercontent.com/1033124/195033535-860fd27c-ab2f-4702-b0d1-c37ab2a5ebae.png)

Reply:
Thanks, nice catch, it was a little confusion